generators/server/index.js

```diff
@@ -522,6 +522,11 @@ module.exports = JhipsterServerGenerator.extend({
             this.camelizedBaseName = _.camelize(this.baseName);
             this.slugifiedBaseName = _.slugify(this.baseName);
             this.lowercaseBaseName = this.baseName.toLowerCase();
+            
+            this.mainClass = _.capitalize(this.camelizedBaseName);
+            if (!this.camelizedBaseName.endsWith('App')) {
+               this.mainClass += 'App';
```

Comment:
We need to do this check for angular app name ad well, currently we always append App to basename for that